extraction_validator/judge.py

```diff
@@ -128,43 +134,68 @@ async def judge_knowledge_graph_batch(
         original_texts: List[str],
         kg_outputs: List[Dict],
         original_prompts: List[str],
-    ) -> List[JudgeReport]:
+        batch_size: int = 50000,
+    ) -> str:
         """
-        Process multiple knowledge graphs in batch using OpenAI API.
-
+        Creates batch files for OpenAI API batch processing.
         Args:
             original_texts: List of source texts
             kg_outputs: List of knowledge graph outputs from LLM
             original_prompts: List of original prompts used for KG extraction
-
+            batch_size: The maximum number of requests per batch file.
         Returns:
-            List of JudgeReports for each input
+            A message indicating the created batch files.
         """
         if len(original_texts) != len(kg_outputs) != len(original_prompts):
             raise ValueError("All input lists must have the same length")
 
-        tasks = []
+        batch_files = []
+        batch_ids = []
+        all_requests = []
         for i, (text, kg_output, prompt) in enumerate(
             zip(original_texts, kg_outputs, original_prompts)
         ):
-            task = self._judge_single_graph(text, kg_output, prompt, i)
-            tasks.append(task)
-
-        results = await asyncio.gather(*tasks, return_exceptions=True)
-
-        # Handle exceptions
-        reports = []
-        for i, result in enumerate(results):
-            if isinstance(result, Exception):
-                reports.append(
-                    self._create_error_report(
-                        f"Error processing item {i}: {str(result)}"
-                    )
-                )
-            else:
-                reports.append(result)
-
-        return reports
+            validation_prompt = self._create_validation_prompt(text, kg_output, prompt)
+            request = {
+                "custom_id": f"request-{i}",
+                "method": "POST",
+                "url": "/v1/chat/completions",
+                "body": {
+                    "model": "gpt-4-turbo-preview",
+                    "messages": [
+                        {
+                            "role": "system",
+                            "content": "You are KG-Judge, a precise and rigorous auditor for knowledge graphs. Always return valid JSON in the exact format requested.",
+                        },
+                        {"role": "user", "content": validation_prompt},
+                    ],
+                    "temperature": 0.1,
+                    "max_tokens": 4000,
+                    "response_format": {"type": "json_object"},
+                },
+            }
+            all_requests.append(request)
+
+        for i in range(0, len(all_requests), batch_size):
+            batch = all_requests[i : i + batch_size]
+            batch_file_name = f"batch_requests_{i // batch_size + 1}.jsonl"
```

Comment:
This is fine, but it would be more pythonic to wrap the iterator with `enumerate` instead of calculated the i with `i // batch_size`

```suggestion
        for i, batch_start in enumerate(range(0, len(all_requests), batch_size)):
            batch = all_requests[batch_start : batch_start + batch_size]
            batch_file_name = f"batch_requests_{i+1}.jsonl"
```